bin/fileloadhelpers/FileLoadHelpers.cpp

```diff
@@ -0,0 +1,140 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#include "stdafx.h"
+#include "Codex/Utf8Codex.h"
+
+HRESULT FileLoadHelpers::LoadScriptFromFile(LPCSTR filename, LPCWSTR& contents, bool* isUtf8Out, LPCWSTR* contentsRawOut, UINT* lengthBytesOut, bool printFileOpenError)
+{
+    HRESULT hr = S_OK;
+    LPCWSTR contentsRaw = nullptr;
+    byte * pRawBytes = nullptr;
+    UINT lengthBytes = 0;
+    bool isUtf8 = false;
+    contents = nullptr;
+    FILE * file;
+
+    //
+    // Open the file as a binary file to prevent CRT from handling encoding, line-break conversions,
+    // etc.
+    //
+    if (fopen_s(&file, filename, "rb") != 0)
+    {
+        if (printFileOpenError)
+        {
+#ifdef _WIN32
+            DWORD lastError = GetLastError();
+            char16 wszBuff[512];
+            fprintf(stderr, "Error in opening file '%s' ", filename);
+            wszBuff[0] = 0;
+            if (FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM,
+                nullptr,
+                lastError,
+                0,
+                wszBuff,
+                _countof(wszBuff),
+                nullptr))
+            {
+                fwprintf(stderr, _u(": %s"), wszBuff);
+            }
+            fwprintf(stderr, _u("\n"));
+#elif defined(_POSIX_VERSION)
+            fprintf(stderr, "Error in opening file: ");
+            perror(filename);
+#endif            
+            IfFailGo(E_FAIL);
+        }
+        else
+        {
+            return E_FAIL;
+        }
+    }
+
+    //
+    // Determine the file length, in bytes.
+    //
+    fseek(file, 0, SEEK_END);
+    lengthBytes = ftell(file);
+    fseek(file, 0, SEEK_SET);
+    contentsRaw = (LPCWSTR)HeapAlloc(GetProcessHeap(), 0, lengthBytes + sizeof(WCHAR));
+    if (nullptr == contentsRaw)
+    {
+        fwprintf(stderr, _u("out of memory"));
+        IfFailGo(E_OUTOFMEMORY);
+    }
+
+    //
+    // Read the entire content as a binary block.
+    //
+    fread((void*)contentsRaw, sizeof(char), lengthBytes, file);
+    fclose(file);
+    *(WCHAR*)((byte*)contentsRaw + lengthBytes) = _u('\0'); // Null terminate it. Could be LPCWSTR.
```

Comment:
@akroshg  Same goes to here. 
RE: `fread may not read lengthBytes. Its return value tells the total number of bytes and it can be lower than lengthBytes` 

The usages of `fread` are wrong as commented on this PR. See also https://msdn.microsoft.com/en-us/library/kt0etdcs.aspx or http://man7.org/linux/man-pages/man3/fread.3.html